source/build.py

```diff
@@ -0,0 +1,169 @@
+                self.pruneThisLang.add('en')
+                self.pruneThisLang.add('ja')
+
+    def handle_starttag(self, tag, attrs):
+        prunePos = self.getpos()
+        if self.targetIsFullHTML and tag == 'body':
+            # mark for injection of Google Tag Manager Script.
+            self.results.add(('injectTagManager', prunePos))
+
+        if ( tag == 'a' and (('href', '..') in attrs) and
+                            (('class', '') in attrs)
+           ):
+                #mark for removal: unelegant blank TOC item.
+                self.results.add(('prune', prunePos))
+
+        for attr in attrs:
+            if ((
+                'ja' in self.pruneThisLang and
+                tag == 'a' and
+                attr[0] == 'href' and
+                ('/ja/' in attr[1] or attr[1].startswith('ja/'))
+            ) or (
+                'en' in self.pruneThisLang and
+                tag == 'a' and
+                attr[0] == 'href' and
+                ('/en/' in attr[1] or attr[1].startswith('en/'))
+            )):
+                #mark for removal: TOC items of the other language.
+                self.results.add(('prune', prunePos))
+
+                # retroactively prevent removal of main links from top level page to
+                # each language's index page.
+                if (
+                    not('/en/' in self.targetFullPath or '/ja/' in self.targetFullPath) and
+                    self.targetFullPath.endswith('/index.html') and
+                    (attr[1] == 'en/' or attr[1] == 'ja/')
+                ):
+                    self.results.remove(('prune', prunePos))
+
+
+# ============================================================================
+
+
+if ('mkdocs.yml' in lsResults) and ('docs' in lsResults) and isMkdocsInstalled:
+    print('All dependencies present. Initiate build...')
+    subprocess.call('mkdocs build --clean', shell=True)
+    print('Mkdocs build successful.')
+    siteDirectory = cwd + "/site"
+    htmlFileFullPaths = set([])
+
+    print('Crawling files...')
+    for root, subDirectories, files in os.walk(siteDirectory):
+        for filename in os.listdir(root):
+            if filename.endswith('.html'):
+                htmlFileFullPaths.add(root + '/' + filename)
+                logging.debug(root + '/' + filename + ' found!')
+
+    for htmlFileFullPath in htmlFileFullPaths:
+        logging.debug('accessing ' + htmlFileFullPath)
+        readHtmlFile = open(htmlFileFullPath, 'r')
+        htmlString = readHtmlFile.read()
+        parser = PostMkdocsParser(htmlFileFullPath)
+        parser.feed(htmlString)
+
+        resultsSet = parser.results.copy()
+        readHtmlFile.close()
+        parser.__kill__()
+
+        if resultsSet:
+            htmlLines = htmlString.splitlines()
+            resultsList = []
+            for result in resultsSet:
+                resultsList.append(result)
+            resultsList.sort()
+            resultsList.reverse()
```

Comment:
I'm not sure why this sort and reverse are needed.
Having a comment here would be helpful